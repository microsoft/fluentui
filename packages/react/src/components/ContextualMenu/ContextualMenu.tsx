import * as React from 'react';
import { styled } from '../../Utilities';
import { IContextualMenuProps, IContextualMenuStyleProps, IContextualMenuStyles } from './ContextualMenu.types';
import { ContextualMenuBase } from './ContextualMenu.base';
import { getStyles } from './ContextualMenu.styles';

function onRenderSubMenu(subMenuProps: IContextualMenuProps) {
  return <LocalContextualMenu {...subMenuProps} />;
}

// This is to prevent cyclic import with ContextualMenu.base.tsx.
const LocalContextualMenu: React.FunctionComponent<IContextualMenuProps> = styled<
  IContextualMenuProps,
  IContextualMenuStyleProps,
  IContextualMenuStyles
>(ContextualMenuBase, getStyles, () => ({ onRenderSubMenu }), { scope: 'ContextualMenu' });

/**
 * ContextualMenu description
 */
export const ContextualMenu: React.FunctionComponent<IContextualMenuProps> = LocalContextualMenu;
ContextualMenu.displayName = 'ContextualMenu';
