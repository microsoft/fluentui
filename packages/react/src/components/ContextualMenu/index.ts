export * from './ContextualMenu';
export * from './ContextualMenu.base';
// eslint-disable-next-line deprecation/deprecation
export { IContextualMenuClassNames, IMenuItemClassNames } from './ContextualMenu.classNames';
export * from './ContextualMenu.types';
export * from './ContextualMenuItem';
export * from './ContextualMenuItem.base';
export * from './ContextualMenuItem.types';
export { getMenuItemStyles } from './ContextualMenu.cnstyles';
