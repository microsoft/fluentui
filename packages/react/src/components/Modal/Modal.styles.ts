import { IModalStyleProps, IModalStyles } from './Modal.types';
import { AnimationVariables, getGlobalClassNames, ZIndexes } from '../../Styling';

export const animationDuration = AnimationVariables.durationValue2;

const globalClassNames = {
  root: 'ms-Modal',
  main: 'ms-Dialog-main',
  scrollableContent: 'ms-Modal-scrollableContent',
  isOpen: 'is-open',
  layer: 'ms-Modal-Layer',
};

export const getStyles = (props: IModalStyleProps): IModalStyles => {
  const {
    className,
    containerClassName,
    scrollableContentClassName,
    isOpen,
    isVisible,
    hasBeenOpened,
    modalRectangleTop,
    theme,
    topOffsetFixed,
    isModeless,
    layerClassName,
    isDefaultDragHandle,
    windowInnerHeight,
  } = props;
  const { palette, effects, fonts } = theme;

  const classNames = getGlobalClassNames(globalClassNames, theme);

  return {
    root: [
      classNames.root,
      fonts.medium,
      {
        backgroundColor: 'transparent',
        position: isModeless ? 'absolute' : 'fixed',
        height: '100%',
        width: '100%',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        opacity: 0,
        pointerEvents: 'none',
        transition: `opacity ${animationDuration}`,
      },
      topOffsetFixed &&
        typeof modalRectangleTop === 'number' &&
        hasBeenOpened && {
          alignItems: 'flex-start',
        },
      isOpen && classNames.isOpen,
      isVisible && {
        opacity: 1,
        pointerEvents: 'auto',
      },
      className,
    ],
    main: [
      classNames.main,
      {
        boxShadow: effects.elevation64,
        borderRadius: effects.roundedCorner2,
        backgroundColor: palette.white,
        boxSizing: 'border-box',
        position: 'relative',
        textAlign: 'left',
        outline: '3px solid transparent',
        maxHeight: 'calc(100% - 32px)',
        maxWidth: 'calc(100% - 32px)',
        minHeight: '176px',
        minWidth: '288px',
        overflowY: 'auto',
        zIndex: isModeless ? ZIndexes.Layer : undefined,
      },
      topOffsetFixed &&
        typeof modalRectangleTop === 'number' &&
        hasBeenOpened && {
          top: modalRectangleTop,
        },
      isDefaultDragHandle && {
        cursor: 'move',
      },
      containerClassName,
    ],
    scrollableContent: [
      classNames.scrollableContent,
      {
        overflowY: 'auto',
        flexGrow: 1,
        maxHeight: '100vh',
        selectors: {
          ['@supports (-webkit-overflow-scrolling: touch)']: {
            maxHeight: windowInnerHeight,
          },
        },
      },
      scrollableContentClassName,
    ],
    layer: isModeless && [
      layerClassName,
      classNames.layer,
      {
        position: 'static',
        width: 'unset',
        height: 'unset',
      },
    ],
    keyboardMoveIconContainer: {
      position: 'absolute',
      display: 'flex',
      justifyContent: 'center',
      width: '100%',
      padding: '3px 0px',
    },
    keyboardMoveIcon: {
      // eslint-disable-next-line deprecation/deprecation
      fontSize: fonts.xLargePlus.fontSize,
      width: '24px',
    },
  };
};
