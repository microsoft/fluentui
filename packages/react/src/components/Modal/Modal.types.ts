import * as React from 'react';
import { ResponsiveMode } from '../../ResponsiveMode';
import type { IStyle, ITheme } from '../../Styling';
import type { ILayerProps } from '../../Layer';
import type { IAccessiblePopupProps } from '../../common/IAccessiblePopupProps';
import type { IRefObject, IStyleFunctionOrObject } from '../../Utilities';
import type { IIconProps } from '../../Icon';
import type { IContextualMenuProps } from '../../ContextualMenu';
import type { IOverlayProps } from '../../Overlay';

export interface IDragOptions {
  /**
   * Optional selector for the element where the drag can be initiated. If not supplied when
   * isDraggable is true dragging can be initiated by the whole contents of the modal
   */
  dragHandleSelector?: string;

  /**
   * IconProps for the icon used to indicate that the dialog is in keyboard move mode
   */
  keyboardMoveIconProps?: IIconProps;

  /**
   * The text to use for the modal move menu item
   */
  moveMenuItemText: string;

  /**
   * The text to use for the modal close menu item
   */
  closeMenuItemText: string;

  /**
   * The Draggable Control Menu so that the draggable zone can be moved via the keyboard
   */
  menu: React.FunctionComponent<IContextualMenuProps>;

  /**
   * Whether the draggable content should be prevented from going off-screen
   */
  keepInBounds?: boolean;
}

/**
 * {@docCategory Modal}
 */
export interface IModal {
  /**
   * Sets focus on the first focusable, or configured, child in focus trap zone
   */
  focus: () => void;
}

/**
 * {@docCategory Modal}
 */
export interface IModalProps extends React.RefAttributes<HTMLDivElement>, IAccessiblePopupProps {
  children?: React.ReactNode;

  /**
   * Optional ref to access the `IModal` interface. Use this instead of `ref` for accessing
   * public API of the component.
   */
  componentRef?: IRefObject<IModal>;

  /**
   * Call to provide customized styling that will layer on top of the variant rules.
   */
  styles?: IStyleFunctionOrObject<IModalStyleProps, IModalStyles>;

  /**
   * Theme provided by higher-order component.
   */
  theme?: ITheme;

  /**
   * Whether the dialog is displayed.
   * @defaultvalue false
   */
  isOpen?: boolean;

  /**
   * Whether the overlay is dark themed.
   * @defaultvalue true
   */
  isDarkOverlay?: boolean;

  /**
   * A callback function for when the Modal is dismissed light dismiss, before the animation completes.
   */
  onDismiss?: (ev?: React.MouseEvent<HTMLButtonElement | HTMLElement>) => any;

  /**
   * A callback function which is called after the Modal is dismissed and the animation is complete.
   */
  onDismissed?: () => any;

  /**
   * The specified responsiveMode value for Modal to use.
   * @defaultvalue ResponsiveMode.small
   */
  responsiveMode?: ResponsiveMode;

  /**
   * Defines an optional set of props to be passed through to Layer
   */
  layerProps?: ILayerProps;

  /**
   * Defines an optional set of props to be passed through to Overlay
   */
  overlay?: IOverlayProps;

  /**
   * Whether the dialog can be light dismissed by clicking outside the dialog (on the overlay).
   * @defaultvalue false
   */
  isBlocking?: boolean;

  /**
   * Whether the dialog should be modeless (e.g. not dismiss when focusing/clicking outside of the dialog).
   * if true: isBlocking is ignored, there will be no overlay (isDarkOverlay is ignored),
   * isClickableOutsideFocusTrap is true, and forceFocusInsideTrap is false
   */
  isModeless?: boolean;

  /**
   * Determines the ARIA role of the dialog (alertdialog/dialog)
   * If this is set, it will override the ARIA role determined by isBlocking and isModeless
   *
   * For more information regarding dialogs please see https://w3c.github.io/aria-practices/#alertdialog
   */
  isAlert?: boolean;

  /**
   * Optional class name to be added to the root class
   */
  className?: string;

  /**
   * Optional override for container class
   */
  containerClassName?: string;

  /**
   * Optional override for scrollable content class
   */
  scrollableContentClassName?: string;

  /**
   * A callback function for when the Modal content is mounted on the overlay layer
   * @deprecated Use `layerProps.onLayerDidMount` instead
   */
  onLayerDidMount?: () => void;

  /**
   * ARIA id for the title of the Modal, if any
   */
  titleAriaId?: string;

  /**
   * ARIA id for the subtitle of the Modal, if any
   */
  subtitleAriaId?: string;

  /**
   * Whether the modal should have top offset fixed once opened and expand from the bottom only
   * when the content changes dynamically.
   */
  topOffsetFixed?: boolean;

  /**
   * The options to make the modal draggable
   */
  dragOptions?: IDragOptions;

  /**
   * Allow body scroll on content and overlay on touch devices. Changing after mounting has no effect.
   * @defaultvalue false
   */
  allowTouchBodyScroll?: boolean;

  /**
   * Puts aria-hidden=true on all non-ancestors of the current modal, for screen readers.
   * This is an experimental feature that will be graduated to default behavior after testing.
   * This flag will be removed with the next major release.
   */
  enableAriaHiddenSiblings?: boolean;
}

/**
 * {@docCategory Modal}
 */
export type IModalStyleProps = Required<Pick<IModalProps, 'theme'>> &
  Pick<
    IModalProps,
    'className' | 'containerClassName' | 'scrollableContentClassName' | 'topOffsetFixed' | 'isModeless'
  > & {
    /** Modal open state. */
    isOpen?: boolean;
    /** Modal visible state. */
    isVisible?: boolean;
    /** Modal has been opened state. */
    hasBeenOpened?: boolean;
    /** Positioning of modal on first render */
    modalRectangleTop?: number;
    /** Classname for layer element */
    layerClassName?: string;
    /** Whether this modal is draggable and using the default handler */
    isDefaultDragHandle?: boolean;
    /** The windows inner height */
    windowInnerHeight?: number;
  };

/**
 * {@docCategory Modal}
 */
export interface IModalStyles {
  root: IStyle;
  main: IStyle;
  scrollableContent: IStyle;
  layer: IStyle;
  keyboardMoveIconContainer: IStyle;
  keyboardMoveIcon: IStyle;
}
