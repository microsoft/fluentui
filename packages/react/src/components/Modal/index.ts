export * from './Modal';
export * from './Modal.base';
export * from './Modal.types';
export { IAccessiblePopupProps } from '../../common/IAccessiblePopupProps';
