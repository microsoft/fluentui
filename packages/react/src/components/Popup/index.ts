export * from './Popup';
export * from './Popup.types';
