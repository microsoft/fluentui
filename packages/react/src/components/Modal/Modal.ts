import * as React from 'react';
import { styled } from '../../Utilities';
import { IModalProps, IModalStyleProps, IModalStyles } from './Modal.types';
import { ModalBase } from './Modal.base';
import { getStyles } from './Modal.styles';

export const Modal: React.FunctionComponent<IModalProps> = styled<IModalProps, IModalStyleProps, IModalStyles>(
  ModalBase,
  getStyles,
  undefined,
  {
    scope: 'Modal',
    fields: ['theme', 'styles', 'enableAriaHiddenSiblings'],
  },
);
Modal.displayName = 'Modal';
