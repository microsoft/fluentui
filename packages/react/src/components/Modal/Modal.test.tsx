import * as React from 'react';
import { Modal } from './Modal';
import { ContextualMenu } from '../../ContextualMenu';
import * as path from 'path';
import { isConformant } from '../../common/isConformant';
import { safeCreate } from '@fluentui/test-utilities';
import { resetIds } from '../../Utilities';
import { Popup } from '../Popup/Popup';

describe('Modal', () => {
  beforeEach(() => {
    resetIds();
  });
  afterAll(() => {
    resetIds();
  });

  isConformant({
    Component: Modal,
    displayName: 'Modal',
    requiredProps: { isOpen: true },
    componentPath: path.join(__dirname, 'Modal.ts'),
    disabledTests: ['component-handles-classname'],
  });

  it('renders Modal correctly', () => {
    // Mock createPortal to capture its component hierarchy in snapshot output.
    const ReactDOM = require('react-dom');
    ReactDOM.createPortal = jest.fn(element => {
      return element;
    });

    safeCreate(
      <Modal isOpen={true} className={'test-className'} containerClassName={'test-containerClassName'}>
        Test Content
      </Modal>,
      component => {
        expect(component.toJSON()).toMatchSnapshot();
        ReactDOM.createPortal.mockClear();
      },
    );
  });

  it('renders Modeless Modal correctly', () => {
    // Mock createPortal to capture its component hierarchy in snapshot output.
    const ReactDOM = require('react-dom');
    ReactDOM.createPortal = jest.fn(element => {
      return element;
    });
    safeCreate(
      <Modal
        isOpen={true}
        isModeless={true}
        className={'test-className'}
        containerClassName={'test-containerClassName'}
      >
        Test Content
      </Modal>,
      component => {
        expect(component!.toJSON()).toMatchSnapshot();
        ReactDOM.createPortal.mockClear();
      },
    );
  });

  it('renders Draggable Modal correctly', () => {
    // Mock createPortal to capture its component hierarchy in snapshot output.
    const ReactDOM = require('react-dom');
    ReactDOM.createPortal = jest.fn(element => {
      return element;
    });

    safeCreate(
      <Modal
        isOpen={true}
        isModeless={true}
        className={'test-className'}
        containerClassName={'test-containerClassName'}
        dragOptions={{
          moveMenuItemText: 'Move',
          closeMenuItemText: 'Close',
          menu: ContextualMenu,
        }}
      >
        Test Content
      </Modal>,
      component => {
        expect(component!.toJSON()).toMatchSnapshot();
        ReactDOM.createPortal.mockClear();
      },
    );
  });

  it('renders a Modal with ARIA role alertDialog when isAlert is true ', () => {
    // Mock createPortal to capture its component hierarchy in snapshot output.
    const ReactDOM = require('react-dom');
    ReactDOM.createPortal = jest.fn(element => {
      return element;
    });

    safeCreate(
      <Modal isOpen={true} isAlert={true} className={'test-className'} containerClassName={'test-containerClassName'}>
        Test Content
      </Modal>,
      component => {
        const componentInstance = component.root;
        expect(componentInstance.findByType(Popup).props.role).toBe('alertdialog');
        ReactDOM.createPortal.mockClear();
      },
    );
  });

  it('renders Modal with ARIA role dialog when isModeless and isBlocking are set to true', () => {
    // Mock createPortal to capture its component hierarchy in snapshot output.
    const ReactDOM = require('react-dom');
    ReactDOM.createPortal = jest.fn(element => {
      return element;
    });

    safeCreate(
      <Modal
        isOpen={true}
        isAlert={false}
        isModeless={true}
        isBlocking={true}
        className={'test-className'}
        containerClassName={'test-containerClassName'}
      >
        Test Content
      </Modal>,
      component => {
        const componentInstance = component.root;
        expect(componentInstance.findByType(Popup).props.role).toBe('dialog');
        ReactDOM.createPortal.mockClear();
      },
    );
  });

  it('renders Modal with ARIA role dialog when isAlert is false', () => {
    // Mock createPortal to capture its component hierarchy in snapshot output.
    const ReactDOM = require('react-dom');
    ReactDOM.createPortal = jest.fn(element => {
      return element;
    });

    safeCreate(
      <Modal
        isOpen={true}
        isAlert={false}
        isBlocking={true}
        className={'test-className'}
        containerClassName={'test-containerClassName'}
      >
        Test Content
      </Modal>,
      component => {
        const componentInstance = component.root;
        expect(componentInstance.findByType(Popup).props.role).toBe('dialog');
        ReactDOM.createPortal.mockClear();
      },
    );
  });

  it('renders Modal with ARIA role alertdialog when isBlocking is true', () => {
    // Mock createPortal to capture its component hierarchy in snapshot output.
    const ReactDOM = require('react-dom');
    ReactDOM.createPortal = jest.fn(element => {
      return element;
    });

    safeCreate(
      <Modal
        isOpen={true}
        isBlocking={true}
        className={'test-className'}
        containerClassName={'test-containerClassName'}
      >
        Test Content
      </Modal>,
      component => {
        const componentInstance = component.root;
        expect(componentInstance.findByType(Popup).props.role).toBe('alertdialog');
        ReactDOM.createPortal.mockClear();
      },
    );
  });
});
