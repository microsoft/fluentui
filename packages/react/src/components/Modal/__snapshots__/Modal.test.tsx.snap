// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders Draggable Modal correctly 1`] = `
<span
  className="ms-layer"
>
  <div
    className=
        ms-Fabric
        ms-Layer-content
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: #323130;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          visibility: visible;
        }
        & button {
          font-family: inherit;
        }
        & input {
          font-family: inherit;
        }
        & textarea {
          font-family: inherit;
        }
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDoubleClick={[Function]}
    onDrag={[Function]}
    onDragEnd={[Function]}
    onDragEnter={[Function]}
    onDragExit={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onInput={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onSubmit={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <div
      aria-modal={false}
      onKeyDown={[Function]}
      role="dialog"
      style={
        Object {
          "outline": "none",
          "overflowY": undefined,
        }
      }
    >
      <div
        className=
            ms-Modal
            is-open
            test-className
            {
              -moz-osx-font-smoothing: grayscale;
              -webkit-font-smoothing: antialiased;
              align-items: center;
              background-color: transparent;
              display: flex;
              font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
              font-size: 14px;
              font-weight: 400;
              height: 100%;
              justify-content: center;
              opacity: 1;
              pointer-events: auto;
              position: absolute;
              transition: opacity 0.267s;
              width: 100%;
            }
      >
        <div
          className=
              ms-Dialog-main
              test-containerClassName
              {
                background-color: #ffffff;
                border-radius: 2px;
                box-shadow: 0 25.6px 57.6px 0 rgba(0, 0, 0, 0.22), 0 4.8px 14.4px 0 rgba(0, 0, 0, 0.18);
                box-sizing: border-box;
                cursor: move;
                max-height: calc(100% - 32px);
                max-width: calc(100% - 32px);
                min-height: 176px;
                min-width: 288px;
                outline: 3px solid transparent;
                overflow-y: auto;
                position: relative;
                text-align: left;
                z-index: 1000000;
              }
          id="ModalFocusTrapZone0"
          onBlurCapture={[Function]}
          onFocusCapture={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "transform": "translate(0px, 0px)",
            }
          }
        >
          <div
            aria-hidden={true}
            data-is-visible={true}
            style={
              Object {
                "pointerEvents": "none",
                "position": "fixed",
              }
            }
            tabIndex={0}
          />
          <div
            className=
                ms-Modal-scrollableContent
                {
                  flex-grow: 1;
                  max-height: 100vh;
                  overflow-y: auto;
                }
                @supports (-webkit-overflow-scrolling: touch){& {
                  max-height: 768px;
                }
            data-is-scrollable={true}
          >
            Test Content
          </div>
          <div
            aria-hidden={true}
            data-is-visible={true}
            style={
              Object {
                "pointerEvents": "none",
                "position": "fixed",
              }
            }
            tabIndex={0}
          />
        </div>
      </div>
    </div>
  </div>
</span>
`;

exports[`Modal renders Modal correctly 1`] = `
<span
  className="ms-layer"
>
  <div
    className=
        ms-Fabric
        ms-Layer-content
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: #323130;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          visibility: visible;
        }
        & button {
          font-family: inherit;
        }
        & input {
          font-family: inherit;
        }
        & textarea {
          font-family: inherit;
        }
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDoubleClick={[Function]}
    onDrag={[Function]}
    onDragEnd={[Function]}
    onDragEnter={[Function]}
    onDragExit={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onInput={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onSubmit={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <div
      aria-modal={true}
      onKeyDown={[Function]}
      role="dialog"
      style={
        Object {
          "outline": "none",
          "overflowY": undefined,
        }
      }
    >
      <div
        className=
            ms-Modal
            is-open
            test-className
            {
              -moz-osx-font-smoothing: grayscale;
              -webkit-font-smoothing: antialiased;
              align-items: center;
              background-color: transparent;
              display: flex;
              font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
              font-size: 14px;
              font-weight: 400;
              height: 100%;
              justify-content: center;
              opacity: 1;
              pointer-events: auto;
              position: fixed;
              transition: opacity 0.267s;
              width: 100%;
            }
        role="document"
      >
        <div
          aria-hidden={true}
          className=
              ms-Overlay
              ms-Overlay--dark
              {
                -moz-osx-font-smoothing: grayscale;
                -webkit-font-smoothing: antialiased;
                background-color: rgba(0,0,0,.4);
                bottom: 0px;
                font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                font-size: 14px;
                font-weight: 400;
                left: 0px;
                position: absolute;
                right: 0px;
                top: 0px;
              }
              @media screen and (-ms-high-contrast: active), (forced-colors: active){& {
                border: 1px solid WindowText;
                opacity: 0;
              }
        />
        <div
          className=
              ms-Dialog-main
              test-containerClassName
              {
                background-color: #ffffff;
                border-radius: 2px;
                box-shadow: 0 25.6px 57.6px 0 rgba(0, 0, 0, 0.22), 0 4.8px 14.4px 0 rgba(0, 0, 0, 0.18);
                box-sizing: border-box;
                max-height: calc(100% - 32px);
                max-width: calc(100% - 32px);
                min-height: 176px;
                min-width: 288px;
                outline: 3px solid transparent;
                overflow-y: auto;
                position: relative;
                text-align: left;
              }
          id="ModalFocusTrapZone0"
          onBlurCapture={[Function]}
          onFocusCapture={[Function]}
        >
          <div
            aria-hidden={true}
            data-is-visible={true}
            style={
              Object {
                "pointerEvents": "none",
                "position": "fixed",
              }
            }
            tabIndex={0}
          />
          <div
            className=
                ms-Modal-scrollableContent
                {
                  flex-grow: 1;
                  max-height: 100vh;
                  overflow-y: auto;
                }
                @supports (-webkit-overflow-scrolling: touch){& {
                  max-height: 768px;
                }
            data-is-scrollable={true}
          >
            Test Content
          </div>
          <div
            aria-hidden={true}
            data-is-visible={true}
            style={
              Object {
                "pointerEvents": "none",
                "position": "fixed",
              }
            }
            tabIndex={0}
          />
        </div>
      </div>
    </div>
  </div>
</span>
`;

exports[`Modal renders Modeless Modal correctly 1`] = `
<span
  className="ms-layer"
>
  <div
    className=
        ms-Fabric
        ms-Layer-content
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: #323130;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          visibility: visible;
        }
        & button {
          font-family: inherit;
        }
        & input {
          font-family: inherit;
        }
        & textarea {
          font-family: inherit;
        }
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDoubleClick={[Function]}
    onDrag={[Function]}
    onDragEnd={[Function]}
    onDragEnter={[Function]}
    onDragExit={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onInput={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onSubmit={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <div
      aria-modal={false}
      onKeyDown={[Function]}
      role="dialog"
      style={
        Object {
          "outline": "none",
          "overflowY": undefined,
        }
      }
    >
      <div
        className=
            ms-Modal
            is-open
            test-className
            {
              -moz-osx-font-smoothing: grayscale;
              -webkit-font-smoothing: antialiased;
              align-items: center;
              background-color: transparent;
              display: flex;
              font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
              font-size: 14px;
              font-weight: 400;
              height: 100%;
              justify-content: center;
              opacity: 1;
              pointer-events: auto;
              position: absolute;
              transition: opacity 0.267s;
              width: 100%;
            }
      >
        <div
          className=
              ms-Dialog-main
              test-containerClassName
              {
                background-color: #ffffff;
                border-radius: 2px;
                box-shadow: 0 25.6px 57.6px 0 rgba(0, 0, 0, 0.22), 0 4.8px 14.4px 0 rgba(0, 0, 0, 0.18);
                box-sizing: border-box;
                max-height: calc(100% - 32px);
                max-width: calc(100% - 32px);
                min-height: 176px;
                min-width: 288px;
                outline: 3px solid transparent;
                overflow-y: auto;
                position: relative;
                text-align: left;
                z-index: 1000000;
              }
          id="ModalFocusTrapZone0"
          onBlurCapture={[Function]}
          onFocusCapture={[Function]}
        >
          <div
            aria-hidden={true}
            data-is-visible={true}
            style={
              Object {
                "pointerEvents": "none",
                "position": "fixed",
              }
            }
            tabIndex={0}
          />
          <div
            className=
                ms-Modal-scrollableContent
                {
                  flex-grow: 1;
                  max-height: 100vh;
                  overflow-y: auto;
                }
                @supports (-webkit-overflow-scrolling: touch){& {
                  max-height: 768px;
                }
            data-is-scrollable={true}
          >
            Test Content
          </div>
          <div
            aria-hidden={true}
            data-is-visible={true}
            style={
              Object {
                "pointerEvents": "none",
                "position": "fixed",
              }
            }
            tabIndex={0}
          />
        </div>
      </div>
    </div>
  </div>
</span>
`;
